index.js

```diff
@@ -5,6 +5,18 @@ function fixCapitalisedPlural(fn) {
   };
 }
 
+function fixChangePlural(fn) {
+  return function(str) {
+    const matches = str.match(/[A-Z_][a-z0-9]*$/);
+    const index = matches ? matches.index : 0;
+    const hasUnderscore = matches && matches[0].startsWith("_");
+    const splitIndex = hasUnderscore ? index + 1 : index;
+    const prefix = str.substr(0, splitIndex);
+    const word = str.substr(splitIndex);
+    return `${prefix}${fn(word)}`;
```

Comment:
Just in case the previous function wasn't bound and needs reference to the other inflectors, we should be careful to call it with the new `this` value:
```suggestion
    return `${prefix}${fn.call(this, word)}`;
```